[Do not merge] - More Mod commands

TitanVJ opened the pull request:
# More Mod commands !!!
## Commands:

## PROPAGATEMUTE
* Summary
  * The command spreads permissions for the @Muted role throughout all channels in the guild
  * Channels are created every now and then, but permissions are always setup. This command takes care of monotony of manually setting each channels permissions
* Args
  * No arguments
* Returns
  * An update messages posted to the channel command was called in
* Aliases
  * propm
* Usage
  * `.propagatemute`: Spreads the @MUTED permissions in all channels
  * `.propm`: Spreads the @MUTED permissions in all channels

## SLOWMODE
* Summary
  * When a channel gets out of hand use this command to enter it into slowmode
  * Mods are immune to slowmode. Specifically if you have one of the following perms: Manage Channel, Manage Messages, Administrator Permission, or you are the Server Owner
* Args
  * *Optional argument*: You can specify the amount of seconds between messages to set in slowmode. `Default value is 10 seconds`. 
* Returns
  * No returns
* Aliases
  * N/A
* Usage
  * `.slowmode`: Sets slowmode to 10 second delay between messages 
  * `.slowmode 11`: Sets slowmode to 11 second delay between messages
  * `.slowmode 0`: Disables slowmode in the channel

## MAKECHANNEL
* Summary
  * Makes a channel with all the right permissions
* Args
  * Channel name: duh. No spaces in the names, only hyphens if needed.
  * *Optional argument*: After the name enter "1" if you want to make a secret channel. Secret channel is only visible to mods/execs/other guild dictators.
* Returns
  * Sends a message to #council channel informing them of the new channel and which who called the command
* Aliases
  * N/A
* Usage
  * `.makechannel cmpt420`: Makes a channel called `cmpt420` with all the right permissions
  * `.makechannel deep-council 1`: Makes a channel called `deep-council` which will be setup as a secret channel only visible to mods/execs/etc

## CLEAR
* Summary
  * The command clears the last 10 messages in the channel regardless of user who sent it. The messages cannot be deleted if they are more than 2 weeks old, it'll return an error messages if so. 
* Args
  * *Optional argument*: You can specify the number of messages to clear. Number must be between 1 and 100 inclusively. `Default value is 10`.
* Returns
  * A messages indicating the deletion of messages. The message self destructs after 5 seconds
  * Error message if messages to delete are older than 2 weeks. The message self destruct after 5 seconds
* Aliases
  * N/A
* Usage
  * `.clear`: Clears the last 10 messages in the current channel
  * `.clear 25`: Clears the last 25 messages in the current channel

## PURGE
* Summary
  * Purges (or deletes) the last X messages from a specific user in the current channel
* Args
  * User to purge messages from: You **MUST** @ mention the user
  * *Optional argument*: You can specify the number of messages to purge from teh user. `Default value is 10`.
* Returns
  * Sends a message to current channel indicating command completion. Message self destructs after 5 seconds
* Aliases
  * N/A
* Usage
  * `.purge @someone`: Purges the last 10 messages from @someone in the current channel
  * `.purge @someone 25`: Purges the last 25 messages from @somone in the current channel

## MUTE
* Summary
  * Adds the @MUTED role to a user in order to mute them in the guild
* Args
  * User to mute: You **MUST** @ mention the user to mute
* Returns
  * Sends a pm to the person who's been muted
  * Sends a message to #council to inform them of who was muted and by whom
* Aliases
  * N/A
* Usage
  * `.mute @someone`: Adds the @MUTED role to @someone and pm's them to let them know they've been muted

## UNMUTE
* Summary
  * Unmute a muted user
* Args
  * User to mute: You **MUST** @ mention the user to mute
* Returns
  * Messages the user they've been unmuted and to not repeat the actions that got them muted
  * Error message of @'d user wasn't muted
  * Sends a message to council informing them about the unmuted user and who unmuted them
* Aliases
  * N/A
* Usage
  * `.unmute @someone`: Removes the @MUTED role from @someone if they had it

## LOCK
* Summary
  * If a channel has gotten too out of hand a simple slowmode won't do then it might be time to lock the channel. Locking a channel means it becomes read only for all normal user until it's unlocked
* Args
  * N/A
* Returns
  * Sends a message in the current channel to inform everyone of its locked status
  * Sends a message to council informing them of which channel was locked and who locked it
* Aliases
  * N/A
* Usage
  * `.lock`: This locks the current channel and sends a message to council

## UNLOCK
* Summary
  * Unlocks a locked channel and sends message to council
* Args
  * N/A
* Returns
  * Sends a message in the current channel to inform everyone of its unlocked status
  * Sends a message to council to inform them of the unlocked channel and who unlocked it
  * Sends an error message if the current channel isn't locked
* Aliases
  * N/A
* Usage
  * `.unlock`: Unlocks the current channel if it's locked

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -79,5 +79,390 @@ def __init__(self, bot):
         await ctx.send(embed=eObj)
 
+    @commands.command(aliases=['propm'])
+    async def propigatemute(self, ctx):
```
spelling mistake

TitanVJ replied:
fixed

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -79,5 +79,390 @@ def __init__(self, bot):
+
+        MUTED_ROLE = discord.utils.get(ctx.guild.roles, name='Muted')
+
+        ignoreChannels = [
```
are you sure you will never use these in another command? cause it might be better to assign them to the bot itself rather than just make it an array inside one command

TitanVJ replied:
moved to settings.py

modernNeo commented:
that is my first check over the command, i didnt look too much into it. but also note that you didnt update the help.json file with the command info. please do that too

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -79,5 +79,390 @@ def __init__(self, bot):
+
+        # Tell user of their new freedom and to not abuse it
+        logger.info('[Mod unmute()] letting {} know they\'ve been unmuted and to not do something stupid again')
+        eObj = em(description='You\'ve been unmuted. Don\'t do whatever you did to get muted in the first place again, or else next time you\'ll get more than a ban', footer='Moderator action')
```
`or else next time you'll get more than a ban`: is `ban` meant here or `mute`?

TitanVJ replied:
nice catch. wast supposed to say ban

modernNeo commented:
@TitanVJ will finish reviewing this massive PR this weekend

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -79,5 +80,376 @@ def __init__(self, bot):
+            await channel.delete_messages(messages)
+            logger.info('[Mod clear()] messages from {} deleted'.format(channel))
+        except discord.HTTPException:
+            eObj = em(description='Messagaes cannot be older than 2 weeks', footer='Command Error')
```
`'Messagaes'`

TitanVJ replied:
fixed

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -79,5 +80,376 @@ def __init__(self, bot):
+        logger.info('[Mod mute()] informing {} that they are muted in dm'.format(user))
+        eObj = em(title='Moderator action was taken against you', colour=0xff0000, description='You\'ve been muted in the CSSS server. Message a minion to learn why and how to be unmuted', footer='Moderator action')
+        await user.send(embed=eObj)
+        #TODO add something ^ to react to msg council for reconsideration slash reach out to u
```
What's to be done? If I'm interpreting correctly, some kind of avenue of appeal?
Will take a 2nd look and hopefully approve this weekend.

TitanVJ replied:
meant to remove this. its done now.

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,375 @@ def __init__(self, bot):
+                await channel.set_permissions(MUTED_ROLE, overwrite=overwrite)
+
+        eObj = em(description='Muted permissions spread though all channels like herpies. Enjoy :)', footer='Moderator action')
+        await ctx.send(embed=eObj, delete_after=3.0)
```
this needs to be redone as `eObj` can not potentially be `None`

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,375 @@ def __init__(self, bot):
+
+        # Verify args
+        if not name: 
+            eObj = em(description='Missing arguments', footer='Error in makechannel command')
```
same issue with `eObj` here

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,375 @@ def __init__(self, bot):
+        overwrite = {
+            ctx.guild.default_role : discord.PermissionOverwrite(mention_everyone=False),
+            MUTED_ROLE : discord.PermissionOverwrite()
+        }
```
how come you dont just cover `mention_everyone` below with `setattr`  along with the other functions?

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,375 @@ def __init__(self, bot):
+        logger.info('[Mod slowmode()] slowmode enable on channel: ' + str(ctx.message.channel) + ', time between messages set to ' + str(time))
+
+    @commands.command()
+    async def makechannel(self, ctx, name, secret = 0):
```
got this error when using it
2019-03-31 02:46:57 = WARNING -   File "/usr/src/app/commands_to_load/Mod.py", line 179, in makechannel
    ch = await ctx.guild.create_text_channel(name, overwrites=overwrite)

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/guild.py", line 670, in create_text_channel
    data = await self._create_channel(name, overwrites, ChannelType.text, category, reason=reason, **options)

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/guild.py", line 573, in _create_channel
    'id': target.id

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING - AttributeError: 'NoneType' object has no attribute 'id'

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING - 
The above exception was the direct cause of the following exception:


2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING - Traceback (most recent call last):

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/bot.py", line 898, in invoke
    await ctx.command.invoke(ctx)

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/core.py", line 611, in invoke
    await injected(*ctx.args, **ctx.kwargs)

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/core.py", line 71, in wrapped
    raise CommandInvokeError(exc) from exc

2019-03-31 02:46:57 = WARNING - 
2019-03-31 02:46:57 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'NoneType' object has no attribute 'id'

2019-03-31 02:46:57 = WARNING -

wnfldchen replied:
(See my review for the cause I've found)

modernNeo commented:
@TitanVJ  a big issue with your code is its using the helper_embed in a wrong way, please work on that

modernNeo commented:
@TitanVJ please also link this in the code as a comment, be useful for future reference
https://support.discordapp.com/hc/en-us/articles/360016150952-Slowmode-Slllooowwwiiinng-down-your-channel

TitanVJ commented:
@wnfldchen @modernNeo done all the changes. had to add commands.Cog to all the class declartions so the cogs would load. need to test locally. 

also the @modernNeo the makechannel command works for me locally without issue. so i dont know wtf is happening on the test server.

wnfldchen commented:
I'm looking at the makechannel issue. It looks like there is some role in the overwrite which is not on the test server.

wnfldchen commented:
The issue is because there is no Muted role on the test server. So this bug will appear if, say, the Muted role was accidentally deleted on the CSSS server.

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+            await self.rekt(ctx)
+            return
+
+        MUTED_ROLE = discord.utils.get(ctx.guild.roles, name='Muted')
```
See this line and all other identical lines, `MUTED_ROLE` can be none. Perhaps explore creating the Muted role if it does not already exist, either on command execution or on startup, or enforcing the precondition that the Muted role must exist already created manually on the server, by outputting some visible error message if it doesn't.

modernNeo replied:
@TitanVJ I concur, do a check to ensur that `MUTED_ROLE` does exist and return an error if it doesn't

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+            return
+
+        # Get the MUTED role
+        MUTED_ROLE = discord.utils.get(ctx.guild.roles, name='Muted')
```
(Same as other change)

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        logger.info('[Mod unmute()] user found through mention: {}'.format(user))
+
+        # Get muted role
+        MUTED_ROLE = discord.utils.get(ctx.guild.roles, name='Muted')
```
(Same as other change)

wnfldchen commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        logger.info('[Mod mute()] user found through mention: {}'.format(user))
+
+        # Grab the Muted role
+        MUTED_ROLE = discord.utils.get(ctx.guild.roles, name='Muted')
```
(Same as other change)

wnfldchen commented:
In the meantime I have created a `Muted` role on the test server so testing can proceed. Note that I don't know what server permissions are supposed to be on the `Muted` role and ever since the changes made to the CSSS server perms I can't check which ones are needed, so someone will have to adjust those.

TitanVJ commented:
as far as im concerned this bot is built bespoke for the csss server so i write all my code with that in mind and dont bother considering its use in another server setup. with that said i also make sure the test server is setup the same as the csss server; thats why theres a council channel in there rn, but i guess i forgot to make sure the muted role was added. 

all that said ill add checks for the muted role, so if someone deletes it bugs wont happen

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        numOfMsgs = int(numOfMsgs)
+
+        # Verify args
+        if numOfMsgs > 100 or numOfMsgs < 1: 
```
you ought to do a return at the end of this `if` statement, otherwise, it'll still continue despite being given a bad range

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        except discord.HTTPException:
+            eObj = await em(ctx, description='Messages cannot be older than 2 weeks', footer='Command Error')
+            if eObj is not False:
+                await ctx.send(embed=eObj, delete_after=10.0)
```
any particulare reason why this embed is being deleted after 10 seconds but the `Number of messages to be between 1 and 100 inclusively` embed has no apparent expiration date?

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+
+    @commands.command()
+    async def purge(self, ctx, *args):
+        # Deletes 10 message in channel from user
```
why 10?

modernNeo replied:
update this to mention that a different number can be passed in, but 10 is the default

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        # Verify arguments
+        logger.info('[Mod purge()] verifying arguments')
+        mentions = ctx.message.mentions
+        if len(mentions) != 1: 
```
this should be changed into 2 separate conditional branches, one where it deals with no @s and one with it deals with too many users @ed

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+                return False
+            
+        # Call channel.purge() limit at 100 and bulk = True
+        deleted = await ctx.channel.purge(limit=100, check=check, bulk=True)
```
why place a limit of 100?

TitanVJ replied:
originally i put that thinking nobody would deleted anywhere near that number but i've changed it so it while loops until the number is reached or theres no more messages to delete from the user.

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        # If there get user
+        logger.info('[Mod mute()] checking for mention')
+        mentions = ctx.message.mentions
+        if len(mentions) != 1:
```
separate the logic as mentioned above

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,398 @@ def __init__(self, bot):
+        # Verify mention
+        logger.info('[Mod unmute()] checking for mention')
+        mentions = ctx.message.mentions
+        if len(mentions) != 1:
```
separate the logic as mentioned above

TitanVJ commented:
@wnfldchen fixed it. i removed something i shouldnt have but i've put it back now.

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+            if channel.id not in adminChannels:
+                await channel.set_permissions(MUTED_ROLE, overwrite=overwrite)
+
+        eObj = await em(ctx, description='Muted permissions spread though all channels like herpies. Enjoy :)', footer='Moderator action')
```
I dont like this `Muted permissions spread though all channels like herpies.` if you want to put this in the comments of the code, thats more OK with me but making a joke like that might turn some people off of the discord channel which defeats the purpose of trying to be friendly.

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+            await council.send(embed=eObj)
+
+    @commands.command()
+    async def unmute(self, ctx):
```
encountered this issue

2019-04-16 03:28:58 = INFO - [Mod unmute()] telling council of the unmuting
2019-04-16 03:28:58 = WARNING - Traceback (most recent call last):
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/core.py", line 62, in wrapped
.    ret = await coro(*args, **kwargs)
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/src/app/commands_to_load/Mod.py", line 448, in unmute
.    await council.send(embed=eObj)
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/abc.py", line 745, in send
.    embed = embed.to_dict()
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING - AttributeError: 'coroutine' object has no attribute 'to_dict'
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -
The above exception was the direct cause of the following exception:
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING - Traceback (most recent call last):
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/bot.py", line 898, in invoke
.    await ctx.command.invoke(ctx)
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/core.py", line 611, in invoke
.    await injected(*ctx.args, **ctx.kwargs)
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING -   File "/usr/local/lib/python3.5/site-packages/discord/ext/commands/core.py", line 71, in wrapped
.    raise CommandInvokeError(exc) from exc
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'coroutine' object has no attribute 'to_dict'
2019-04-16 03:28:58 = WARNING -
2019-04-16 03:28:58 = WARNING - /usr/local/lib/python3.5/asyncio/tasks.py:240: RuntimeWarning: coroutine 'embed' was never awaited
.  result = coro.send(None)

modernNeo commented:
@TitanVJ on the test server, when I create a channel not with the bot, it already has the muted role. if that is the case, then why is the command `.propagatemute` needed?

![screenshot 32](https://user-images.githubusercontent.com/14242088/56410293-7194af80-6231-11e9-876b-0a0f646c283a.jpg)

modernNeo commented:
@TitanVJ also, please update the (`documentation/Working_on_the_Bot.md`) file with all the steps that a user needs to take on their own discord guild to be able to run these new commands on their own discord guild as I cannot run `.propagatemute` on my own discord guild and dont have time to investigate what needs to be done

    
    2019-04-19 21:00:38 = INFO - [Mod propagatemute()] invoking message deleted
    2019-04-19 21:00:38 = INFO - overwrite =[<discord.permissions.PermissionOverwrite object at 0x7fbe09012ca8>]
    2019-04-19 21:00:38 = INFO - overwrite =[<discord.permissions.PermissionOverwrite object at 0x7fbe09012ca8>]
    2019-04-19 21:00:38 = WARNING - Traceback (most recent call last):

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 62, in wrapped
        ret = await coro(*args, **kwargs)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e/commands_to_load/Mod.py", line 126, in propagatemute
        await channel.set_permissions(MUTED_ROLE, overwrite=overwrite)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/abc.py", line 592, in set_permissions
        await http.edit_channel_permissions(self.id, target.id, allow.value, deny.value, perm_type, reason=reason)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/http.py", line 210, in request
        raise Forbidden(r, data)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING - discord.errors.Forbidden: FORBIDDEN (status code: 403): Missing Permissions

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING - 
    The above exception was the direct cause of the following exception:


    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING - Traceback (most recent call last):

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/bot.py", line 898, in invoke
        await ctx.command.invoke(ctx)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 611, in invoke
        await injected(*ctx.args, **ctx.kwargs)

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 71, in wrapped
        raise CommandInvokeError(exc) from exc

    2019-04-19 21:00:38 = WARNING - 
    2019-04-19 21:00:38 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: Forbidden: FORBIDDEN (status code: 403): Missing Permissions

    2019-04-19 21:00:38 = WARNING -

wnfldchen commented:
Yeah, looks like youve got to document the correct bot role permissions, if im reading that output correctly

modernNeo commented:
@TitanVJ still got the following error
    
        2019-04-27 15:00:01 = INFO - [Mod propagatemute()] invoking message deleted
        2019-04-27 15:00:01 = INFO - overwrite =[<discord.permissions.PermissionOverwrite object at 0x7fcc2687e5b8>]
        2019-04-27 15:00:01 = INFO - overwrite =[<discord.permissions.PermissionOverwrite object at 0x7fcc2687e5b8>]
        2019-04-27 15:00:01 = WARNING - Traceback (most recent call last):

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 62, in wrapped
            ret = await coro(*args, **kwargs)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e/commands_to_load/Mod.py", line 126, in propagatemute
            await channel.set_permissions(MUTED_ROLE, overwrite=overwrite)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/abc.py", line 592, in set_permissions
            await http.edit_channel_permissions(self.id, target.id, allow.value, deny.value, perm_type, reason=reason)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/http.py", line 210, in request
            raise Forbidden(r, data)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING - discord.errors.Forbidden: FORBIDDEN (status code: 403): Missing Permissions

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING - 
        The above exception was the direct cause of the following exception:


        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING - Traceback (most recent call last):

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/bot.py", line 898, in invoke
            await ctx.command.invoke(ctx)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 611, in invoke
            await injected(*ctx.args, **ctx.kwargs)

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING -   File "/home/jace/NAS_Documents/Git/wall_e_ENV/lib/python3.5/site-packages/discord/ext/commands/core.py", line 71, in wrapped
            raise CommandInvokeError(exc) from exc

        2019-04-27 15:00:01 = WARNING - 
        2019-04-27 15:00:01 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: Forbidden: FORBIDDEN (status code: 403): Missing Permissions

        2019-04-27 15:00:01 = WARNING - 

and this point, I suggest you make a new discord guild and record everything you had to do to make the `.propgatemute` command work and document those steps here.

TitanVJ commented:
@modernNeo i made a new server, added boy with link from documentation. then added the minions and muted roles with default perms since only the names matter and i added the council channel. then i ran the command and it work.

the error is stating missing permissions so can u double check that the bot in your server has all the same permissions as walle in csss server

TitanVJ commented:
dont approve yet, gotta add something for one of the commands at request of council

modernNeo commented:
in those cases, either close the pr or add `[do not merge` to the beginning of the title

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+        logger.info('[Mod slowmode()] invoking message deleted')
+
+        if not ctx.message.author in discord.utils.get(ctx.guild.roles, name="Minions").members:
+            logger.info('[Mod slowmode()] unathoriezed command attempt detected. Being handled.')
```
`unathoriezed`, type?

modernNeo replied:
`unathoriezed`, typo?

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+
+        # Add muted role to user
+        await user.add_roles(MUTED_ROLE)
+        logger.info('[Mod mute()] adding muted role to user')
```
`adding` should be `added`

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+        await channel.set_permissions(MINIONS_ROLE, send_messages=True)
+
+        # Message channel to notify the status
+        logger.info('[Mod lock()] lock message sent to {}'.format(channel))
```
this should go under line 482 or be rephrased as `lock message will be sent to`

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+        await channel.set_permissions(MINIONS_ROLE, overwrite=None)
+
+        # Tell the channel
+        logger.info('[Mod unlock()] unlock message sent to {}'.format(channel))
```
isnt the lock message technically sent on 530, with that being verified on line 531? so this should be below line 532

modernNeo commented on commands_to_load/Mod.py:
```diff
@@ -82,5 +83,460 @@ def __init__(self, bot):
+            await ctx.send(embed=eObj)
+
+        # Tell council
+        logger.info('[Mod unlock()] council informed of unlocked channel')
```
again, this should be on line 540

TitanVJ commented:
gonna rework this. no eta wat so ever.
